### PR TITLE
Surface db config errors

### PR DESCRIPTION
**Pull request by charisk**
Adds logic to capture and surface any errors when loading and reading the database configuration file. At the moment we're not surfacing the actual error messages in the panel - the exact UX is TBD.

## Checklist
N/A - internal only.

- [ ] [CHANGELOG.md](https://github.com/github/vscode-codeql/blob/main/extensions/ql-vscode/CHANGELOG.md) has been updated to incorporate all user visible changes made by this pull request.
- [ ] Issues have been created for any UI or other user-facing changes made by this pull request.
- [ ] _[Maintainers only]_ If this pull request makes user-facing changes that require documentation changes, open a corresponding docs pull request in the [github/codeql](https://github.com/github/codeql/tree/main/docs/codeql/codeql-for-visual-studio-code) repo and add the `ready-for-doc-review` label there.

### CODE REVIEW

File: extensions/ql-vscode/src/databases/ui/db-tree-data-provider.ts [hunk 1]
```diff
@@ -36,14 +36,17 @@ export class DbTreeDataProvider implements TreeDataProvider<DbTreeViewItem> {
...
-    const dbItems = this.dbManager.getDbItems();
+    const dbItemsResult = this.dbManager.getDbItems();
 
-    // Add a sample warning as a proof of concept.
-    const warningTreeViewItem = createDbTreeViewItemWarning(
-      'There was an error',
-      'Fix it'
-    );
+    if (dbItemsResult.isFailure) {
+      const warningTreeViewItem = createDbTreeViewItemError(
```
Comment (shati-patel) [inline extensions/ql-vscode/src/databases/ui/db-tree-data-provider.ts:42]: As noted on the call: `warningTreeViewItem` -> `errorTreeViewItem`

File: extensions/ql-vscode/src/databases/ui/db-tree-view-item.ts [hunk 2]
```diff
@@ -26,10 +26,10 @@ export class DbTreeViewItem extends vscode.TreeItem {
   }
 }
 
-export function createDbTreeViewItemWarning(label: string, tooltip: string): DbTreeViewItem {
+export function createDbTreeViewItemError(label: string, tooltip: string): DbTreeViewItem {
   return new DbTreeViewItem(
     undefined,
-    new vscode.ThemeIcon('warning', new vscode.ThemeColor('problemsWarningIcon.foreground')),
+    new vscode.ThemeIcon('warning', new vscode.ThemeColor('problemsErrorIcon.foreground')),
```
Comment (shati-patel) [inline extensions/ql-vscode/src/databases/ui/db-tree-view-item.ts:32]: Should this be error as well?

```suggestion
    new vscode.ThemeIcon('error', new vscode.ThemeColor('problemsErrorIcon.foreground')),
```

**Review by shati-patel — APPROVED** [review]
A couple of minor comments, but otherwise LGTM! Thanks for explaining ✨

File: extensions/ql-vscode/src/common/value-result.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+export class ValueResult<TValue> {
+  private constructor(
+    private readonly errorMsgs: string[],
+    private readonly val?: TValue,
+  ) {
+  }
+
+  public static ok<TValue>(value: TValue): ValueResult<TValue> {
+    if (value === undefined) {
+      throw Error('Value must be set for successful result');
```
[the remark this answers is not in the document]
Comment (charisk) [reply extensions/ql-vscode/src/common/value-result.ts:13]: It's the same thing - I'll update to use the same in both places!

```suggestion
      throw new Error('Value must be set for successful result');
```

**Pull request merged**